src/features/notifications/NotificationList.tsx

```diff
@@ -1,13 +1,83 @@
-import React from 'react'
+import { groupBy } from 'lodash'
+import React, { useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
+import { FlatList, RefreshControl } from 'react-native'
+import { formatDistance, fromUnixTime } from 'date-fns'
+import Box from '../../components/Box'
 import Text from '../../components/Text'
+import { Notification } from '../../store/account/accountSlice'
+import NotificationGroup from './NotificationGroup'
+import NotificationShow from './NotificationShow'
 
-const NotificationList = () => {
+type Props = {
+  notifications: Notification[]
+  refreshing: boolean
+  onRefresh: () => void
+}
+const NotificationList = ({ notifications, refreshing, onRefresh }: Props) => {
   const { t } = useTranslation()
+  const [groupedNotifications, setGroupedNotifications] = useState<
+    Array<Array<Notification>>
+  >([])
+  const [
+    selectedNotification,
+    setSelectedNotification,
+  ] = useState<Notification | null>(null)
+
+  useEffect(() => {
+    // TODO: Figure out how to group these.
+    // Currently, they are grouped by the date-fns function formatDistance
+    // Then, sorted newest to oldest
+    // Seems good, but need to confirm
+
+    const now = new Date()
+
+    const grouped = groupBy(notifications, (notification) =>
+      formatDistance(fromUnixTime(notification.time), now, {
+        addSuffix: true,
+      }),
+    )
+
+    const arr = Object.keys(grouped)
+      .map((k) => grouped[k])
+      .sort((a) => a[0].time)
+
+    setGroupedNotifications(arr)
+  }, [notifications])
+
+  // TODO: Pull to refresh or polling updates??
+
   return (
-    <Text variant="h3" marginVertical="m">
-      {t('notifications.list.title')}
-    </Text>
+    <Box flex={1} alignContent="space-between">
+      <Text variant="h3" marginVertical="m" flexGrow={1} paddingHorizontal="l">
+        {t('notifications.list.title')}
+      </Text>
+
+      <FlatList
+        refreshControl={
+          <RefreshControl
+            refreshing={refreshing}
+            onRefresh={onRefresh}
+            tintColor="white"
+          />
+        }
+        refreshing={refreshing}
+        style={{ flexGrow: 0 }}
+        data={groupedNotifications}
+        keyExtractor={(item, idx) => `${item[0].id}.${idx}`}
+        renderItem={({ item, index }) => (
+          <NotificationGroup
+            index={Math.min(groupedNotifications.length - 1, 10) - index}
+            notifications={item}
+            onNotificationSelected={setSelectedNotification}
+          />
+        )}
+      />
+      <NotificationShow
```

Comment:
I think in a future PR we could make sure of holy-sheet for these as well so that you'd get them animated up from the bottom, a scrollview interior, and swipe down to close.

Reply:
That was my initial thought, but the design is a little different here than the rest of the app. There's a margin around the card and a blurred background. Whatever the case, I think I need to have a conversation with Pete around his vision for all of this.

Reply:
yeah I was thinking about adding a blurred background option directly in that library. and I think you should be able to override the styles around the container to give it some margin

Reply:
happy to take a pass on that at some point in the future